pkg/controller/pxc/backup.go

```diff
@@ -73,73 +74,58 @@ func (r *ReconcilePerconaXtraDBCluster) reconcileBackups(cr *api.PerconaXtraDBCl
 				return fmt.Errorf("storage %s doesn't exist", bcp.StorageName)
 			}
 
-			bcpjob, err := bcpObj.Scheduled(&bcp, strg, operatorPod)
-			if err != nil {
-				return fmt.Errorf("unable to schedule backup: %w", err)
-			}
-			err = setControllerReference(cr, bcpjob, r.scheme)
-			if err != nil {
-				return fmt.Errorf("set owner ref to backup %s: %v", bcp.Name, err)
-			}
-
-			// Check if this Job already exists
-			currentBcpJob := new(batchv1beta1.CronJob)
-			err = r.client.Get(context.TODO(), types.NamespacedName{
-				Name:      bcpjob.Name,
-				Namespace: bcpjob.Namespace,
-			}, currentBcpJob)
-			if err != nil && !k8serrors.IsNotFound(err) {
-				return errors.Wrapf(err, "create scheduled backup %s", bcp.Name)
+			sch := BackupScheduleJob{}
+			schRaw, ok := r.crons.backupJobs.Load(bcp.Name)
+			if ok {
+				sch = schRaw.(BackupScheduleJob)
 			}
 
-			if k8serrors.IsNotFound(err) {
-				err = r.client.Create(context.TODO(), bcpjob)
+			if !ok || sch.PXCScheduledBackupSchedule.Schedule != bcp.Schedule ||
+				sch.PXCScheduledBackupSchedule.StorageName != bcp.StorageName {
+				r.log.Info("Creating or updating backup job", "name", bcp.Name, "schedule", bcp.Schedule)
+				r.deleteBackupJob(bcp.Name)
+				jobID, err := r.crons.crons.AddFunc(bcp.Schedule, r.createBackupJob(cr, bcp, strg.Type))
 				if err != nil {
-					return errors.Wrapf(err, "create scheduled backup %s", bcp.Name)
-				}
-			} else if !reflect.DeepEqual(currentBcpJob.Spec, bcpjob.Spec) {
-				err = r.client.Update(context.TODO(), bcpjob)
-				if err != nil {
-					return errors.Wrapf(err, "update backup schedule %s", bcp.Name)
+					return errors.Wrap(err, "failed to add backup cron job")
```

Comment:
Should all following jobs be failed if one has the wrong schedule format?

Reply:
81a9c2fd